target/linux/ath79/dts/ar7161_ubnt_routerstation.dtsi

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7100.dtsi"
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	aliases {
+		led-boot = &rf;
+		led-failsafe = &rf;
+		led-running = &rf;
+		led-upgrade = &rf;
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-output-names = "ref";
+		clock-frequency = <40000000>;
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		rf: rf_green {
+			label = "ubnt:green:rf";
+			gpios = <&gpio 2 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+		debounce-interval = <60>;
+
+		wps {
+			label = "sw4";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&rbc 0x17>;
+};
+
+&eth1 {
+	status = "okay";
+
+	mtd-mac-address = <&rbc 0x17>;
+	mtd-mac-address-increment = <1>;
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "ecoscentric,redboot-fis-partitions";
+		};
+
+		rbc: partition@fff000 {
```

Comment:
Yeah, that way of mixing old syntax with new one to trigger two parsers (`ecoscentric,redboot-fis-partitions` and `fixed-partitions`) seems really hacky to me. Don't do that as you're trying to describe the same thing in two conflicting ways at the same time.

So it has to be either: `ecoscentric,redboot-fis-partitions` or `fixed-partitions`.

I guess `ecoscentric,redboot-fis-partitions` is needed because flash layout isn't fixed. So go with that and then either:
1. Use user-space (as suggested by @mkresin) to read MAC
2. OR Rewrite `redboot-fis-partitions` to allow specifying nodes

The first solution should be much easier.